docs: Possible update to injection detection (#1144)
NVIDIA/NeMo-Guardrails: docs/user-guides/guardrails-library.md

| Docker     | 2057  | 115 |
| In-Process | 3227  | 157 |

### Injection Detection

> NeMo Guardrails offers detection of potential injection attempts such as code injection, cross-site scripting, SQL injection, and template injection.

Reviewer: ```suggestion
NeMo Guardrails offers detection of potential exploitation attempts via injection such as code injection, cross-site scripting, SQL injection, and template injection.
```

Injection detection is primarily intended to be used in agentic systems to enhance other security controls as part of a defense-in-depth strategy.

The first part of injection detection is [YARA rules](https://yara.readthedocs.io/en/stable/index.html).
A YARA rule specifies a set of strings--text or binary patterns--to match and a Boolean expression that specifies the logic of the rule.

> YARA rules is a technology that is familiar to many security teams.

Reviewer: ```suggestion
YARA rules are a technology that is familiar to many security teams.
```

The second part of injection detection is specifying the action to take when a rule is triggered.
You can specify to *reject* the text and return "I'm sorry, the desired output triggered rule(s) designed to mitigate exploitation of {detections}."

> Or, you can specify to *omit* the triggering text from the response.

Reviewer: I think we should add a note of some kind suggesting that `reject` is by far the safest option and also the most suitable for production deployments.

#### About the Default Rules

By default, NeMo Guardrails provides the following rules:

- Code injection (Python): Recommended if the LLM output is used as an argument to downstream functions or passed to a code interpreter.
- SQL injection: Recommended if the LLM output is used as part of a SQL query to a database.

> - Template injection (Jinja): Recommended for use if LLM output is rendered using templating languages like Jinja.

Reviewer: ```suggestion
- Template injection (Jinja): Recommended for use if LLM output is rendered using templating languages. Currently, only Jinja is supported.
```

The syntax used is specific to Jinja, so even if you're using something Jinja-like, it won't work.

Author: Thanks for the clarification about "Jinja-like" not working.  I'm not a fan of "currently" because, depending on priorities, it might read "currently" for three or four years.  How about...

```
- Template injection (Jinja): Recommended for use if LLM output is rendered using the Jinja templating language.
  This rule is usually paired with code injection rules.
```

When we add more languages, we'll just have to update this section--and would have to even if "currently" was in the text.
